Crosshair, Interaction hints and improvements to projectiles

AxelSchneewind opened the pull request:
Adds a crosshair (a simple dot for now) to the canvas.

In addition, adds a text element next to it that can show text depending on the object the player looks at.

Refactors the prefabs for dropped items and shop item slots such that their colliders are set appropriately. In particular, this makes picking up arrows work as it should.

Further, includes enhancements to projectiles that stick to objects. E.g. arrows can now stick to chests and fall down when the chest is destroyed. Works similarly for enemies where arrows can stick to the bones (maximum number of attached arrows can be configured).

AxelSchneewind committed:
add scripts for crosshair

AxelSchneewind committed:
add crosshair to canvas

AxelSchneewind committed:
make player script update interaction hints and remove interaction zone

AxelSchneewind committed:
implement dynamic interaction hint for trapdoor

AxelSchneewind committed:
remove interaction area

AxelSchneewind committed:
make droppeditem compute its collider from model bounds

AxelSchneewind committed:
simplify fruit prefabs

AxelSchneewind committed:
simplify item slot prefab

AxelSchneewind committed:
simplify bow ammo prefab

AxelSchneewind committed:
Add interaction hint to dropped item prefab

AxelSchneewind committed:
fix arrow rotation

AxelSchneewind committed:
switch projectile hit detection to raycast

This allows for arbitrary speeds without clipping through objects

AxelSchneewind committed:
fix initial arrow position

AxelSchneewind committed:
add scripts to manage projectiles attached to other objects

AxelSchneewind committed:
make opponent target for attaching projectiles

AxelSchneewind committed:
make droppeditem attachable to objects and add rigidbody

AxelSchneewind committed:
split attached objects into base and derived class

AxelSchneewind committed:
simplify chest prefabs

AxelSchneewind committed:
make arrows snap to bones slightly

AxelSchneewind committed:
simplify hit handling of projectile

AxelSchneewind committed:
use mesh collider for shop

AxelSchneewind committed:
reduce initial ammo

AxelSchneewind committed:
fix hitzone

365Bit commented on Assets/Scripts/Items/DroppedItem.cs:
I would add a space between pick up and the item name in brackets.

AxelSchneewind committed:
make interaction hint formatting configurable for dropped item

AxelSchneewind committed:
fix tip position for projectile prefab

AxelSchneewind committed:
handle case where no objects can be attached

AxelSchneewind committed:
improve position computation on arrow hit

Now moves the arrow to the intersection point of
its trajectory and an orthogonal y-plane through
the opponent position.

When hitting other objects, a variable depth factor
is used to compute position.
Can be used to define material softness in the future.
